src/Common/src/Interop/Windows/kernel32/Interop.GetConsoleTitle.cs

```diff
@@ -5,68 +5,66 @@
 using System.Diagnostics;
 using System.Text;
 using System.Runtime.InteropServices;
+using System;
 
 internal partial class Interop
 {
     internal partial class Kernel32
     {
-        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "GetConsoleTitleW")]
-        private static extern int GetConsoleTitle([Out]StringBuilder title, int nSize);
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, SetLastError = true, ExactSpelling = true)]
+        private static extern uint GetConsoleTitleW(ref char title, uint nSize);
 
-        private const int InitialBufferSizeInChars = 256;
+        private static bool s_isWindows7 = GetIsWindows7();
 
-        // Although msdn states that the max allowed limit is 65K,
-        // desktop limits this to 24500 as buffer sizes greater than it
-        // throw.
-        private const int MaxAllowedBufferSizeInChars = 24500;
-
-        // GetConsoleTitle sometimes interprets the second parameter (nSize) as number of bytes and sometimes as the number of chars.
-        // Instead of doing complicated and dangerous logic to determine if this may or may not occur,
-        // we simply assume the worst and reserve a bigger buffer. This way we may use a bit more memory,
-        // but we will always be safe.
-        private static int CharCountToByteCount(int numChars)
+        private static bool GetIsWindows7()
         {
-            return numChars * 2;
+            // Version lies from the OS kick in starting with Windows 8 (6.2)
+            Version version = Environment.OSVersion.Version;
+            return version.Major == 6 && version.Minor == 1;
         }
 
-        // 1. We first try to call the GetConsoleTitle with an InitialBufferSizeInChars of 256.
-        // 2. Then based on the return value either increase the capacity or return failure.
-        internal static int GetConsoleTitle(out string title, out int titleLength)
+        internal static string GetConsoleTitle(out int error)
         {
-            int lastError = 0;
-            StringBuilder sb = new StringBuilder(CharCountToByteCount(InitialBufferSizeInChars + 1));
-            int len = GetConsoleTitle(sb, sb.Capacity);
+            error = Errors.ERROR_SUCCESS;
 
-            if (len <= 0)
-            {
-                lastError = Marshal.GetLastWin32Error();
-            }
-            else if (len > MaxAllowedBufferSizeInChars)
-            {
-                // Title is greater than the allowed buffer so we do not read the title and only pass the length to the caller.
-                title = string.Empty;
-                titleLength = len;
-                return 0;
-            }
-            else
+            Span<char> initialBuffer = stackalloc char[256];
+            ValueStringBuilder builder = new ValueStringBuilder(initialBuffer);
```

Comment:
This is the first place we are using ValueStringBuilder under Common\Interop. We have the buffer allocation logic under the implementation everywhere else.

Reply:
Yeah, I'll move the helper into the calling code to be consistent.